polaris.shopify.com/src/components/Markdown/Markdown.tsx

```diff
@@ -50,14 +50,16 @@ export const CodeVisibilityProvider: ComponentType<
   }>
 > = ({children, showCode, setShowCode}) => {
   // Attempt to inherit from parent context if it's set
-  const [codeVisibleFromContext, setShowCodeFromContext] = useCodeVisibility();
-
+  const {
+    showCode: codeVisibleFromContext,
+    setShowCode: setShowCodeFromContext,
+  } = useCodeVisibility();
   return (
     <CodeVisibilityContext.Provider
-      value={[
-        codeVisibleFromContext ?? showCode,
-        setShowCodeFromContext ?? setShowCode,
-      ]}
+      value={{
+        showCode: typeof codeVisibleFromContext !== undefined ?? showCode,
```

Comment:
Changed this check, as an explicitly declared `false` value should still be inherited. Previous code would ignore all falsey values, including an explicit `false` bool